### PR TITLE
Update to latest PHP 8.1 syntax

### PR DESCRIPTION
**Pull request by samsonasik**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Since composer.json require php 8.1, php 8.1 syntax can be applied.

### CODE REVIEW

**Comment by samsonasik** [comment]
Ready to review 👍

**Comment by froschdesign** [comment]
@samsonasik 
> Ready to merge 👍

You mean ready for review! 😉 
Please also add labels and a milestone if possible.

**Comment by samsonasik** [comment]
@froschdesign updated 👍

**Comment by froschdesign** [comment]
@samsonasik 
Do not forget a label, which will serve for quick categorisation here in the repository and will be used in the release notes.
Example: https://github.com/laminas/laminas-filter/releases/tag/2.39.0

_For all your pull requests!_

**Comment by samsonasik** [comment]
@froschdesign just added 👍

**Review by gsteel — APPROVED** [review]
👍

File: src/FilterChain.php [hunk 1]
```diff
@@ -178,7 +178,7 @@ public function attach($callback, $priority = self::DEFAULT_PRIORITY)
                     get_debug_type($callback)
                 ));
             }
-            $callback = [$callback, 'filter'];
+            $callback = $callback->filter(...);
```
Comment (gsteel) [inline src/FilterChain.php:181]: This causes test failures in `laminas/input-filter` - However, the test case over there is wrong to assert an array when the return type is clearly documented as `callable`

Comment (gsteel) [reply src/FilterChain.php:181]: Ref: https://github.com/laminas/laminas-inputfilter/pull/118

**Pull request merged**